src/inmanta/warnings.py

```diff
@@ -19,47 +19,44 @@
 import logging
 import warnings
 from enum import Enum
-from typing import Dict, List, Mapping, Optional, TextIO, Type, Union
+from typing import Dict, List, Literal, Mapping, Optional, TextIO, Type, Union
 
-
-class InmantaWarning(Warning):
-    """
-    Base class for Inmanta Warnings.
-    """
-
-    def __init__(self, *args: object):
-        Warning.__init__(self, *args)
+REGEX_INMANTA_MODULE: str = r"^(inmanta|inmanta\..*|inmanta_.*)$"
 
 
 class WarningBehaviour(Enum):
-    WARN: str = "default"
-    IGNORE: str = "ignore"
-    ERROR: str = "error"
+    WARN: Literal["default"] = "default"
+    IGNORE: Literal["ignore"] = "ignore"
+    ERROR: Literal["error"] = "error"
 
 
 class WarningRule:
     """
     A single rule for warning handling. Describes the desired behaviour when an error occurs.
-    When type is set, the rule is only applied to subclasses of the warning type.
+
+    :param module: A regex that must match the name of the module generating the warning.
     """
 
-    def __init__(self, action: WarningBehaviour, tp: Optional[Type[InmantaWarning]] = None) -> None:
+    def __init__(self, action: WarningBehaviour, module: Optional[str] = None) -> None:
         self.action: WarningBehaviour = action
-        self.type: Type[Warning] = tp if tp is not None else InmantaWarning
+        self.module: Optional[str] = module
 
     def apply(self) -> None:
-        warnings.filterwarnings(self.action.value, category=self.type)
+        if self.module is not None:
+            warnings.filterwarnings(self.action.value, module=self.module)
+        else:
+            warnings.filterwarnings(self.action.value)
 
 
 class WarningOption:
     """
     An option to manage warnings. Consists of a name and a range of possible values, each tied to a warning rule.
     For example, applying
     WarningOption(
-        "disable-runtime-warnings",
-        {True: WarningRule(WarningBehaviour.IGNORE, CompilerRuntimeWarning)}
+        "disable-inmanta-warnings",
+        {True: WarningRule(WarningBehaviour.IGNORE, module=REGEX_INMANTA_MODULE)}
```

Comment:
We lose the ability here to filter on a specific warning. Couldn't that be useful still? i.e. allow filtering on both module and type?

Reply:
I removed this, because we are not using this functionality and I don't like to have code in our code base that is not used. Apart from that I cannot think of a use cases where we want to have a different setting for one specific type of warnings. Do you have a specific use case in mind?

Reply:
One thing that comes to mind is raising an error for compiler deprecation warnings. Or ignoring a specific known issue. But for a moment I forgot that this can only be configured through the options we explicitly expose, so you're right that we're not using it at the moment. I withdraw my comment, we can always add it again if/when the need arises.